File: Data Structures & Algorithms/duplicate-integer/submission-5.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> mp;
        for(auto num : nums){
            mp[num]++;
        }
        for(auto [key, val] : mp){
            if(val > 1){
                return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/target-sum/submission-5.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        dp = [defaultdict(int) for i in range(len(nums)+1)]
        dp[0][0] = 1
        for i in range(len(nums)):
            for count, val in dp[i].items():
                dp[i+1][count - nums[i]] += val
                dp[i+1][count + nums[i]] += val
        return dp[len(nums)][target]